corehq/apps/locations/models.py

```diff
@@ -505,35 +529,20 @@ def save(self, *args, **kwargs):
 
     @classmethod
     def filter_by_type(cls, domain, loc_type, root_loc=None):
-        loc_id = root_loc._id if root_loc else None
-        relevant_ids = [r['id'] for r in cls.get_db().view('locations/by_type',
-            reduce=False,
-            startkey=[domain, loc_type, loc_id],
-            endkey=[domain, loc_type, loc_id, {}],
-        ).all()]
+        if root_loc:
+            query = root_loc.sql_location.get_descendants(include_self=True)
+        else:
+            query = SQLLocation.objects
+        ids = query.filter(domain=domain, location_type__name=loc_type).ids()
+
         return (
-            cls.wrap(l) for l in iter_docs(cls.get_db(), list(relevant_ids))
+            cls.wrap(l) for l in iter_docs(cls.get_db(), list(ids))
             if not l.get('is_archived', False)
         )
 
     @classmethod
-    def filter_by_type_count(cls, domain, loc_type, root_loc=None):
-        loc_id = root_loc._id if root_loc else None
-        return cls.get_db().view('locations/by_type',
-            reduce=True,
-            startkey=[domain, loc_type, loc_id],
-            endkey=[domain, loc_type, loc_id, {}],
-        ).one()['value']
-
-    @classmethod
     def by_domain(cls, domain, include_docs=True):
-        relevant_ids = set([r['id'] for r in cls.get_db().view(
-            'locations/by_type',
-            reduce=False,
-            startkey=[domain],
-            endkey=[domain, {}],
-        ).all()])
-
+        relevant_ids = SQLLocation.objects.domain(domain).ids()
```

Comment:
Good point about `.ids()`, I'll switch that.

`.filter(domain=domain)` just started to feel really boilerplate-y.  You're right about it being more intuitive though, and half the time you're doing `.filter(domain=domain, foo='bar')` anyways.

Another method I was thinking about adding here is `.couch_locations()`, which would call `.ids()` and pipe that through `iter_docs`.  That should help with the pattern of replacing couch views with SQL, but still being able to use couch.  It feels pretty dirty having a queryset method return couch objects though.  Still, then the next step would be updating the references to use SQLLocations directly.  This'd make it easier to do incrementally.  (I probably won't do that right away though, I was mostly doing this now because I wanted to think through a harder problem while I had Tyler around to bounce ideas off).